src/main/java/com/jfrog/ide/common/tree/DescriptorFileTreeNode.java

```diff
@@ -0,0 +1,39 @@
+package com.jfrog.ide.common.tree;
+
+import java.util.Collection;
+
+public class DescriptorFileTreeNode extends FileTreeNode {
+    public DescriptorFileTreeNode(String filePath) {
+        super(filePath);
+    }
+
+    public void addDependency(Artifact dependency) {
+        Artifact clonedDep = (Artifact) dependency.clone();
+        add(clonedDep);
+        if (clonedDep.getTopSeverity().isHigherThan(topSeverity)) {
+            topSeverity = clonedDep.getTopSeverity();
+        }
+        sortChildren();
```

Comment:
Removes code duplication
```suggestion
        addDependencies(Collections.singletonList(dependency));
```